nemo_aligner/models/nlp/gpt/gpt_sft_model.py

```diff
@@ -88,7 +88,7 @@ def get_loss_and_metrics(self, batch, forward_only):
         set_sync_funcs(self, forward_only)
 
         fwd_bwd_function = get_forward_backward_func()
-        fwd_loss_fn = self.get_forward_output_and_loss_func(forward_only)
+        fwd_loss_fn = self.get_forward_output_and_loss_func(forward_only, tuning=True)
```

Comment:
what does tuning do?

Reply:
It controls the keys that are returned in the batch: https://github.com/NVIDIA/NeMo/blob/b847bf75c371931e4f17ea426741c1d023afa0c0/nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py#L1211-L1222. If `tuning=False`, we don't return the keys that are necessary for sequence packing (and thus CP) in TE.

Note also that `tuning` is set to `True` in NeMo's SFT: https://github.com/NVIDIA/NeMo/blob/b847bf75c371931e4f17ea426741c1d023afa0c0/nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py#L407